examples/hn_scraper.py

```diff
@@ -21,25 +21,20 @@ def next_item(self):
 
 def download_metadata(hn_id):
     # Given an hacker news id returned from the api, fetch metadata
-    req = requests.get(f"https://hacker-news.firebaseio.com/v0/item/{hn_id}.json")
-    if not req.json():
-        logger.warning(f"error getting payload from item {hn_id} trying again")
-        time.sleep(0.5)
-        return download_metadata(hn_id)
-    return req.json()
-
-
-def recurse_tree(metadata):
-    try:
-        parent_id = metadata["parent"]
-        parent_metadata = download_metadata(parent_id)
-        return recurse_tree(parent_metadata)
-    except KeyError:
-        return (metadata["id"], {**metadata, "key_id": metadata["id"]})
+    # Try 3 times, waiting more and more, or give up
+    for i in range(1, 4):
+        req = requests.get(f"https://hacker-news.firebaseio.com/v0/item/{hn_id}.json")
+        if not req.json():
+            logger.warning(f"error getting payload from item {hn_id} trying again")
+            time.sleep(1**i)
+        else:
+            return req.json()
```

Comment:
ahah sorry, I did write it, I saw it was recursively calling the same function over and over and wanted to put a limit to that, so I refactored it to do exponential backoff, and doing that I forgot rule number 1 :smile: Since this request happens on a map step, I can't use the option in SimplePollingSource, but I think it's ok to just skip things that can't be downloaded in the example